x/merkledb/proof.go

```diff
@@ -480,27 +480,71 @@ type KeyChange struct {
 	Value Maybe[[]byte]
 }
 
+// A change proof proves that a set of key-value changes occurred
+// between two trie roots, where each key-value pair's key is
+// between some lower and upper bound (inclusive).
 type ChangeProof struct {
+	// Invariant: At least one of [StartProof], [EndProof], or
```

Comment:
Technically this is only true once https://github.com/ava-labs/avalanchego/pull/1772 is merged but that seems kind of imminent